pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GroovyTransformFunction.java

```diff
@@ -429,13 +408,11 @@ public double[][] transformToDoubleValuesMV(ValueBlock valueBlock) {
 
   @Override
   public String[][] transformToStringValuesMV(ValueBlock valueBlock) {
-    if (_stringValuesMV == null) {
-      _stringValuesMV = new String[DocIdSetPlanNode.MAX_DOC_PER_CALL][];
-    }
+    int length = valueBlock.getNumDocs();
+    initStringValuesMV(length);
```

Comment:
Why this is set as `MAX_DOC_PER_CALL` initially? Will this causing the frequent resizing problem?

Reply:
I'm pretty sure that is a typo (we used to always allocate 10K docs, but then changed it to length)